source/neuropod/bindings/java/src/main/native/utils.h

```diff
@@ -27,5 +29,12 @@ namespace jni
 // Convert jstring to cpp string
 std::string toString(JNIEnv *env, jstring target);
 
+// A wrapper for env->FindClass, will throw a cpp exception if the find fails.
+jclass findClass(JNIEnv *env, const char *name);
+
+// Throw a Java NeuropodJNIException exception. This throw is only valid for the Java side, in the cpp side it is just
+// like a normal function: it won't interupt the normal program work flow.
```

Comment:
interRupt

Reply:
I dont follow, is it going to be used as a "normal" function at all?

Reply:
It is not a throw in the C++ side function, instead it throws an exception in the Java side. When the program counter go back to the java side, the java will know there is an exception thrown.

For example, in the C++ side:

std::vector<int>* a = new std::vector<int>();
delete a;
throwJavaException(env, e.what());
delete a;

This will cause a segfault, because the throwJavaException won't interrupt the C++ side code, both delete is executed.